### ISSUE TITLE
Hold on CORS deploy resolution

### ISSUE THREAD

**Issue by Joelkw** [sourcegraph/sourcegraph#15584]
This is a proxy used by sonarqube, snyk, and link-preview-expander extensions.

Original issue: https://sourcegraph.slack.com/archives/CJX299FGE/p1603480630111300 after this [Github comment](https://github.com/sourcegraph/infrastructure/pull/2209#pullrequestreview-510790952)

Interim solution: #15582  

Upcoming solutions: https://sourcegraph.slack.com/archives/CMT39K56Z/p1604930788227000

Likely not be us who implements the solution.

**Comment by jeanduplessis** [comment]
Progress for unblocking the extensions (sans link preview) can be seen in this PR: https://github.com/sourcegraph/infrastructure/pull/2238

**Comment by jeanduplessis** [comment]
Here's the link to the meeting notes between Felix, Gonzalo, Chayim, and myself: https://docs.google.com/document/d/18AJnJX8AhFEq4j9fJtLJun4Ljv3PbxE0NQVJwqLvVRw/edit

**Comment by Joelkw** [comment]
@jeanduplessis since #15582 happened, we should be able to update the default URLs from the heroku proxy we used to our own proxy url now. I think that should be a quick enough change we could fit it in this iteration?

**Comment by jeanduplessis** [comment]
I spoke with various stakeholders yesterday and the proposed resolution for unblocking the link preview extension is to add an endpoint to the Sourcegraph API. While it's acknowledged this isn't the most ideal solution (which would likely have been having a separate microservice), it is the more pragmatic one with the least short-term downside. We might consider moving this out into a separate service at a later stage.

I will open up a separate issue for implementing this new endpoint. While we might not have any Go expertise in the team, this should be small enough for us to take on as an upskill task for someone in the team, and get support from one of the other backend engineers along the way.

**Comment by jeanduplessis** [comment]
> @jeanduplessis since #15582 happened, we should be able to update the default URLs from the heroku proxy we used to our own proxy url now. I think that should be a quick enough change we could fit it in this iteration? (I didn't strictly define that "resolution" of this issue would include us updating the URL once we create our own proxy, since when I made this issue, we didn't know what the solution might look like, so just wanted to call that out explicitly.)

@Joelkw I've confirmed with Felix that we should be able to fit this change into the current iteration and have opened this issue for the work: https://github.com/sourcegraph/sourcegraph/issues/15794

@Joelkw since we have issues to track both actions required to resolve this issue I'll leave it to you on whether we want to close this one in favor of tracking progress on these:
- #15735
- #15794

**Comment by Joelkw** [comment]
@jeanduplessis That all sounds good to me! Thanks for seeing these through. I think we should close this and track continuing progress on those followup issues.
